### PR TITLE
Kernel+LibKeyboard+ProcFS: Store and expose the current kernel keymap

### LINKED ISSUE #2684: LibKeyboard: API to get current active keymap
It seems like there is no syscall or otherwise implemented API to get the current active keymap - this would be useful for the "Keyboard settings" application, which only shows a blank selection for now:

image

### PR DESCRIPTION
**Pull request by vkoskiv**
This set of patches enables storing the loaded keymap name in the kernel, and then querying it back by reading from `/proc/keymap`

<p align="left">
  
</p>

Once this PR is merged, I can modify the KeyboardSettings program to show the currently selected keymap.
Resolves #2684

### CODE REVIEW

**Comment by linusg** [comment]
Closes #2684 :^)

File: Libraries/LibKeyboard/CharacterMap.h [hunk 1]
```diff
@@ -43,9 +43,13 @@ class CharacterMap {
 
     u32 get_char(KeyEvent);
     void set_character_map_data(CharacterMapData character_map_data);
+    void set_character_map_name(String& character_map_name);
```
Comment (awesomekling) [inline Libraries/LibKeyboard/CharacterMap.h:46]: Please use a `const String&` here, as `set_character_map_name` does not intend to mutate the passed-in `String`.
Also, you can omit the parameter name sine it's already part of the function name :)

File: Libraries/LibKeyboard/CharacterMap.h [hunk 2]
```diff
@@ -43,9 +43,13 @@ class CharacterMap {
...
+
+    String character_map_name();
```
Comment (awesomekling) [inline Libraries/LibKeyboard/CharacterMap.h:48]: This should be `const` qualified, as it does not mutate the state of the `CharacterMap`.

File: Kernel/Syscalls/setkeymap.cpp [hunk 3]
```diff
@@ -27,6 +27,8 @@
 #include <Kernel/Devices/KeyboardDevice.h>
 #include <Kernel/Process.h>
 
+#define CHAR_MAP_NAME_SIZE 50
```
Comment (awesomekling) [inline Kernel/Syscalls/setkeymap.cpp:30]: No need for this to be a macro, instead just use a function-local `constexpr size_t`

File: Kernel/Devices/KeyboardDevice.h [hunk 4]
```diff
@@ -51,7 +51,9 @@ class KeyboardDevice final : public IRQHandler
     KeyboardDevice();
 
     void set_client(KeyboardClient* client) { m_client = client; }
-    void set_maps(Keyboard::CharacterMapData character_map);
+    void set_maps(Keyboard::CharacterMapData character_map, String& character_map_name);
```
Comment (awesomekling) [inline Kernel/Devices/KeyboardDevice.h:54]: Both of these arguments should be const references.

File: Kernel/Devices/KeyboardDevice.h [hunk 5]
```diff
@@ -51,7 +51,9 @@ class KeyboardDevice final : public IRQHandler
...
+
+    String keymap_name() { return m_character_map.character_map_name(); }
```
Comment (awesomekling) [inline Kernel/Devices/KeyboardDevice.h:56]: Please make this `const` qualified.

File: Kernel/FileSystem/ProcFS.cpp [hunk 6]
```diff
@@ -372,6 +374,16 @@ Optional<KBuffer> procfs$interrupts(InodeIdentifier)
     return builder.build();
 }
 
+Optional<KBuffer> procfs$keymap(InodeIdentifier)
+{
+    KBufferBuilder builder;
+    JsonObjectSerializer<KBufferBuilder> json { builder };
+    auto name = KeyboardDevice::the().keymap_name();
+    json.add("keymap", name);
```
Comment (awesomekling) [inline Kernel/FileSystem/ProcFS.cpp:382]: I would lose the temporary here:
```suggestion
    json.add("keymap", KeyboardDevice::the().keymap_name());
```

**Comment by awesomekling** [comment]
Please don't add fixups to the last commit of the PR, add them to the relevant commit.
You can use the fixup feature of `git rebase -i` for this. :)

**Comment by vkoskiv** [comment]
Fixup applied with a bit of rebase magic!

**Comment by awesomekling** [comment]
Ayy nice :) Today we learned about git!

**Pull request merged**